templates/commands/goldentest/record_test.go

```diff
@@ -0,0 +1,166 @@
+// Copyright 2023 The Authors (see AUTHORS file)
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// Package goldentest implements golden test related subcommands.
+package goldentest
+
+import (
+	"context"
+	"path/filepath"
+	"testing"
+
+	"github.com/abcxyz/abc/templates/common"
+	"github.com/abcxyz/pkg/logging"
+	"github.com/abcxyz/pkg/testutil"
+	"github.com/google/go-cmp/cmp"
+)
+
+//nolint:paralleltest // This test cannot be run in parallel
+func TestRecordCommand(t *testing.T) {
+	specYaml := `api_version: 'cli.abcxyz.dev/v1alpha1'
+kind: 'Template'
+
+desc: 'A simple template'
+
+steps:
+  - desc: 'Include some files and directories'
+    action: 'include'
+    params:
+      paths: ['.']
+`
+	testYaml := "api_version: 'cli.abcxyz.dev/v1alpha1'"
+
+	cases := []struct {
+		name                  string
+		testName              string
+		filesContent          map[string]string
+		expectedGoldenContent map[string]string
+		wantErr               string
+	}{
+		{
+			name: "simple_test_succeeds",
+			filesContent: map[string]string{
+				"spec.yaml":                      specYaml,
+				"a.txt":                          "file A content",
+				"b.txt":                          "file B content",
+				"testdata/golden/test/test.yaml": testYaml,
+			},
+			expectedGoldenContent: map[string]string{
+				filepath.Join("test", "test.yaml"): testYaml,
+				filepath.Join("test", "a.txt"):     "file A content",
+				filepath.Join("test", "b.txt"):     "file B content",
+			},
+		},
+		{
+			name: "multiple_tests_succeeds",
+			filesContent: map[string]string{
+				"spec.yaml":                       specYaml,
+				"a.txt":                           "file A content",
+				"testdata/golden/test1/test.yaml": testYaml,
+				"testdata/golden/test2/test.yaml": testYaml,
+			},
+			expectedGoldenContent: map[string]string{
+				filepath.Join("test1", "test.yaml"): testYaml,
+				filepath.Join("test1", "a.txt"):     "file A content",
+				filepath.Join("test2", "test.yaml"): testYaml,
+				filepath.Join("test2", "a.txt"):     "file A content",
+			},
+		},
+		{
+			name: "oudated_golden_file_removed",
+			filesContent: map[string]string{
+				"spec.yaml":                        specYaml,
+				"a.txt":                            "file A content",
+				"testdata/golden/test/test.yaml":   testYaml,
+				"testdata/golden/test/oudated.txt": "oudated file",
```

Comment:
oudated -> outdated, throughout this file

Reply:
Thank you